nearcore/src/runtime/mod.rs

```diff
@@ -823,14 +823,24 @@ impl RuntimeAdapter for NightshadeRuntime {
 
         // Total amount of gas burnt for converting transactions towards receipts.
         let mut total_gas_burnt = 0;
+        let mut total_size = 0u64;
         // TODO: Update gas limit for transactions
         let transactions_gas_limit = gas_limit / 2;
         let mut transactions = vec![];
         let mut num_checked_transactions = 0;
 
         let runtime_config = self.runtime_config_store.get_config(current_protocol_version);
 
-        while total_gas_burnt < transactions_gas_limit {
+        // In general, we limit the number of transactions via send_fees.
+        // However, as a second line of defense, we want to limit the byte size
+        // of transaction as well. Rather than introducing a separate config for
+        // the limit, we compute it heuristically from the gas limit and the
+        // cost of roundtripping a byte of data through disk.
+        let size_limit = transactions_gas_limit
+            / (runtime_config.wasm_config.ext_costs.storage_write_value_byte
+                + runtime_config.wasm_config.ext_costs.storage_read_value_byte);
```

Comment:
Interesting approach, sounds about right. What size limit does it correspond to and how does it relate to a typical chunk size?

Reply:
This is 13megs worth of transactions, which translates to 30meg encoded chunk. I would assume that the *typical* chunk size is much smaller than that, but I don't know how to verify that.

Reply:
Ok thanks! 13MB feels okay in terms of fitting enough actual work in it.

From the other side of things, checking if it is still too large. Am I right that the full chunk needs to be sent to other validators? Because 30MB to send over the network takes >2.4 seconds on a 100Mbit/s link at perfect utilisation. With a 1Gbit/s link, theoretical fastest possible to send it to one other validator is 0.24 seconds. That sounds a bit optimistic to me.

Reply:
Yeah... I don't actually know what's our budget here is, so I am not ready to say what's the actual limit here. 13 megs feels like its roughtly the right order of magnitude (today, chunks can be up to 140megs). I think it's reasonable to use the current code, and adjust downwards if we observe (or test) that this is not enough.